src/jupyterhub_config/hub_singleuser.py

```diff
@@ -0,0 +1,143 @@
+import fcntl
+import grp
+import logging
+import os
+import pwd
+import subprocess
+import tempfile
+from pathlib import Path
+
+
+class SingleUserEnvManager:
+    """
+    Manages the environment and user-specific settings for a single user in a JupyterHub server.
+    """
+
+    def __init__(self,
+                 username,
+                 groupname: str = 'jupyterhub', ):
+        """
+        Initializes the environment manager for the specified user.
+        :param username: The name of the user for whom the environment is set up.
+        :param groupname: The name of the OS system group to which the user belongs.
+        """
+        self.username = username  # OS system username
```

Comment:
I don't understand why the juipyterhub username and the linux username need to be the same

Reply:
somewhere down the line (I think it's auth) will need to check that system user exist with the same jupyterhub username.

Reply:
I don't quite get that - why would the linux username need to match the auth username?

Reply:
maybe for some spawner it's not required. But I don't see it concerns us.

Reply:
https://discourse.jupyter.org/t/is-authentication-without-system-users-possible/1770

Reply:
> But I don't see it concerns us.

If the system user is the same for all containers, then we don't have to worry about the YARN queue stuff

Reply:
but then how are files shared with the same system user and different juptyerhub user?

Reply:
Well, I don't know how jupyterhub works, but my assumption is that it would be like a lot of other software and the system user wouldn't matter. The software would run as the system user but the username associated with the software isn't an existing linux account at all.

Reply:
maybe there are workarounds. Is it worth the time to explore it, absolutely no. Yarn is working fine with different users.

Reply:
> Yarn is working fine with different users.

Not if it's bypassing fair share it isn't

Reply:
I will explore some yarn setting (`spark.yarn.queue`) on the client's side.

Reply:
If you just use the same user for the containers you don't have to worry about this

Reply:
I created another issue for this. https://github.com/kbase/cdm-jupyterhub/issues/97

It's really not how Jupyterhub works as far as I understand.

Reply:
If you're saying that JH requires the linux username to match the JH username then yeah, that won't work and we'll have to look into modifying yarn or how the submit works